### PR TITLE
Add NUST Hackathon page

### LINKED ISSUE #9416: Create a dedicated page for NUST Hackathon

### PR DESCRIPTION
**Pull request by sm1990**
screencapture-localhost-3000-namibia-hackathon-2024-09-18-12_22_20

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: components/nust/challenges/challenges.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+                  <p>Proposal submission deadline: <b>Friday, 27 September 2024 (23:59 CAT)</b>
+                    <br />
+                    Announcement of accepted proposals: <b>Thursday, 3 October 2024</b></p>
+
+
+                  <h3>Round 2: Final Implementation and Demo Submission</h3>
+
+
+                  <p>Participants must submit their final implementations using the Ballerina language, along with a 15-minute demo on <b>Monday, 7 October 2024</b></p>
```
Comment (MaryamZi) [inline components/nust/challenges/challenges.js:75]: on or by?

File: components/nust/challenges/challenges.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        <Row>
+          <Col xs={12} md={6} lg={6} className={styles.boxCol}>
+            <div className={styles.cardWrapper}>
+              <div>
+                <h3>Guidelines for the proposals:</h3>
+                <div className={styles.cardDescription}>
+                  <ul>
+                    <li className={styles.customLi}>Project proposals that demonstrate strong relevance to actual problems and align with the shared interests of the university or the country are better suited.</li>
+                    <li className={styles.customLi}>WSO2, in collaboration with NUST members, will evaluate the proposals and communicate the acceptance to the participants.</li>
+                    <li className={styles.customLi}>Once the idea is accepted, participants will be required to implement it.</li>
```
Comment (MaryamZi) [inline components/nust/challenges/challenges.js:96]: Was the content reviewed separately?

These aren't really guidelines, right? @azinneera

File: components/nust/rewards/Rewards.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <path d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z" />
+              </svg>
+              Rewards
+            </h2>
+          </Col>
+        </Row>
+        <Row>
+          <Col xs={12}>
+            <p>
+              All participants whose proposals are accepted in Round 1 are eligible for a Ballerina branded T-shirt and sticker sheet.
```
Comment (MaryamZi) [inline components/nust/rewards/Rewards.js:50]: T-shirt or t-shirt?

File: components/nust/rules/Rules.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            </h2>
+          </Col>
+        </Row>
+        <Row>
+          <div>
+            <div className={styles.cardDescription}>
+              <div className={styles.content}>
+                <ul className={styles.customUl}>
+                  <li className={styles.customLi}>The applicant must be a legal resident or citizen in Namibia.</li>
+                  <li className={styles.customLi}>The Ballerina programming language should be used for the complete or part of the final implementation.</li>
```
Comment (MaryamZi) [inline components/nust/rules/Rules.js:53]: Isn't this simply

```suggestion
                  <li className={styles.customLi}>The Ballerina programming language should be used in the final implementation.</li>
```

File: components/nust/rules/Rules.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                  <li className={styles.customLi}>The event is open to individuals above the age of 18 who reside in Namibia and any student of NUST even below the required age limit (with the written consent of the guardian/parent).</li>
```
Comment (MaryamZi) [inline components/nust/rules/Rules.js:54]: Can be a bit clearer?

File: components/nust/challenges/challenges.js (hunk 2 above)
Comment (azinneera) [reply components/nust/challenges/challenges.js:96]: The content was reviewed, but seems like this part is missed. @sm1990 Let's change it to "Proposal Selection and Next Steps"

File: components/nust/challenges/challenges.js (hunk 1 above)
Comment (azinneera) [reply components/nust/challenges/challenges.js:75]: on or before

File: components/nust/challenges/challenges.js (hunk 2 above)
Comment (azinneera) [reply components/nust/challenges/challenges.js:96]: @MaryamZi can you review https://docs.google.com/document/d/1sgtiFHlEFYTA0oNmtqX0jzNnLjGsgda5qfbP-0ojcg8/edit

File: components/nust/challenges/challenges.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+              </div>
+            </div>
+          </Col>
+          <Col xs={12} md={6} lg={6} className={styles.boxCol}>
+            <div className={styles.cardWrapper}>
+              <div>
+                <h3>Example technical areas to focus on (but not limited to):</h3>
+                <div className={styles.cardDescription}>
+                  <ul>
+                    <li className={styles.customLi}><b>API Integration Challenge:</b><br /> Participants can choose an existing APIs (e.g., weather data, social media, university LMS) and demonstrate how they can integrate it into a useful application using Ballerina.</li>
```
Comment (azinneera) [inline components/nust/challenges/challenges.js:108]:
```suggestion
                    <li className={styles.customLi}><b>API Integration Challenge:</b><br /> Participants can choose existing APIs (e.g., weather data, social media, university LMS) and demonstrate how they can integrate it into a useful application using Ballerina.</li>
```

File: pages/hacktoberfest/index.js
[the remark this answers is not in the document]
Comment (sm1990) [reply pages/hacktoberfest/index.js:1]: Yes. Just updated some styles related to that.

File: components/nust/challenges/challenges.js (hunk 1 above)
Comment (sm1990) [reply components/nust/challenges/challenges.js:75]: @Ashi1993

File: components/nust/rules/Rules.js (hunk 5 above)
Comment (azinneera) [reply components/nust/rules/Rules.js:54]: Let's break this to a list? 

- Any legal resident or citizen in Namibia above the age of 18
- Any student of NUST even below the required age limit (with the written consent of the guardian/parent)

File: components/nust/rewards/Rewards.js (hunk 3 above)
Comment (azinneera) [reply components/nust/rewards/Rewards.js:50]: T-shirt I believe

**Review by azinneera — APPROVED** [review]

**Review by keizer619 — APPROVED** [review]

**Review by NipunaMadhushan — APPROVED** [review]

**Pull request merged**